### PR TITLE
Convert modding profile posts component to typescript

### PR DESCRIPTION
**Pull request by notbakaneko**

### CODE REVIEW

**Review by nanaya — CHANGES_REQUESTED** [review]

File: resources/js/modding-profile/posts.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            <div className='beatmap-discussion-timestamp__icons-container'>
+              <span className='fas fa-reply' title={trans('common.buttons.reply')} />
+            </div>
+          </div>
+        </div>
+        <div className={discussionClasses}>
+          <div className='beatmap-discussion__discussion'>
+            <Post
+              key={post.id}
+              beatmap={null}
```
[the remark this answers is not in the document]
Comment (notbakaneko) [reply resources/js/modding-profile/posts.tsx:69]: It's explicitly selected beatmap or none selected, not optional for the places that use the prop

[the remark this answers is not in the document]
Comment (nanaya) [reply resources/js/modding-profile/posts.tsx:69]: `null` would mean the post doesn't have beatmap_id (an "all" section post) except it's not always the case here...

[the remark this answers is not in the document]
Comment (notbakaneko) [reply resources/js/modding-profile/posts.tsx:68]: It's not optional and shouldn't be #10291

[the remark this answers is not in the document]
Comment (nanaya) [reply resources/js/modding-profile/posts.tsx:68]: with the current usage it ended up being optional but either way it's wrong I guess 🤷

**Review by nanaya — APPROVED** [review]

**Pull request merged**